### ISSUE TITLE
v2 release roadmap

### ISSUE THREAD

**Issue by mars3142** [fayeed/dash_chat#117]
I see in the projects tab, the planned features for v2, but there is nothing about timing. Would it be possible to push you v2-branch, so everybody can see any progress or can help with that? 

For example, I want to implement an image loader, because I would use Firebase or S3 for image storage and won't save the fix url with the message, because it can be changed anytime. So I just want to add the filename and the ChatMessage will do the magic to transform it into any valid url. Also you mentioned composer actions in the README.md file, and I would like to use that as well.

Forking and building it 100% by myself is not an option, because I don't know you current state with that.

**Comment by fayeed** [comment]
@mars3142 I have not started working on the v2 yet, I waiting for @SebastienBtr to get some free time we will then start working on the v2 then, it could be 2 months or so before we start.

As for the implementation that you want to add I think that can be done with the current feature set just try with `messageImageBuilder` it should return the image name that you set, then in image builder, you could use the `getDownloadURL` to get the URL for the image and render your custom widget. An even better solution would be that you could just get the URL in when you upload the image & set it in the `ChatMessage`.

Hope this answers your question.

**Comment by mars3142** [comment]
After writing this feature request, I found the `messageImageBuilder` parameter and it was the missing link for me for images. 

And because of v2 roadmap. I have to wait until everyone can test/fix/help with that. Good to know.

You can close the issue or leave it open as a reminder ;)

**Comment by elgansayer** [comment]
Didn't realise there was a v2 coming.  I had a fair few issues with Dash due to performance and started to roll a complete UI system including a functioning backend. With all the features in the world. It's mostly done. The design uses overrides instead of builders.
Anyway!
Maybe we could collaborate?  I think ill be a week or three yet and have to build a while backend stream system example yet. 

![image](https://user-images.githubusercontent.com/6216372/89665401-8c7cae80-d8d0-11ea-8aa1-c838cbfe6cfe.png)

**Comment by fayeed** [comment]
@elgansayer I have been planning it for a while now, so finally decided to build it, if you are interested in collaborating I have a rough spec draft for v2 you can check it on [google docs](https://docs.google.com/document/d/1kJfj3dy1DsIPAGF7ADMt5h38p9qKNTzJWu0LA58fj7M/edit?usp=sharing), I going to share it here for discussions later once it is finalized.

**Comment by elgansayer** [comment]
@fayeed Very interesting. Perhaps I could collab, at least the test and backend system could be donated. 

A fair bit of that list is already done and some others are very good ideas and some others are beyond the scope of my chat system.

I deleted the comment originally since I thought it may be a bit rude to say :D Hope that was ok.

**Comment by fayeed** [comment]
@elgansayer No problem man, once I have finalized the spec I will create a new issue to discuss & share ideas with others & possibly look for more contributors too. If you have any ideas or suggestions feel free to share.

**Comment by fayeed** [comment]
@elgansayer @mars3142 I have pinned #127 for discussion if you have some feedback you can let us know there.
